File: msr/reference/go/remove-duplicates.go
package main

func remove_duplicates(xs []int) []int {
	seen := map[int]bool{}
	out := []int{}
	for _, x := range xs {
		if !seen[x] {
			seen[x] = true
			out = append(out, x)
		}
	}
	return out
}

File: msr/reference/zig/remove-duplicates.zig
const std = @import("std");

pub fn remove_duplicates(allocator: std.mem.Allocator, xs: []const i64) ![]i64 {
    const out = try allocator.alloc(i64, xs.len);
    var n: usize = 0;
    for (xs) |x| {
        if (std.mem.indexOfScalar(i64, out[0..n], x) == null) {
            out[n] = x;
            n += 1;
        }
    }
    return out[0..n];
}
